src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double COEFFICIENT = 1.5;
+    private static final int LIST_CAPACITY = 10;
+    private T[] data = (T[]) new Object[LIST_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (size == data.length) {
+            this.data = (T[]) grow();
+        }
+        data[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        isIndexInBoundAdd(index);
+        if (size == data.length) {
```

Comment:
I suggest to rename method grow to growIfFull and add size theck to renamed method, as well as this.data reassinging